File: app/src/main/java/com/penseapp/acaocontabilidade/chat/chats/presenter/ChatsReaderPresenterImpl.java
package com.penseapp.acaocontabilidade.chat.chats.presenter;

import com.penseapp.acaocontabilidade.chat.chats.view.ChatsAdapterView;
import com.penseapp.acaocontabilidade.chat.chats.interactor.ChatsReaderInteractor;
import com.penseapp.acaocontabilidade.chat.chats.interactor.ChatsReaderInteractorImpl;
import com.penseapp.acaocontabilidade.chat.chats.model.Chat;

/**
 * Created by unity on 10/12/16.
 */

public class ChatsReaderPresenterImpl implements ChatsReaderPresenter {

    private final ChatsAdapterView chatsView;
    private final ChatsReaderInteractor chatsReaderInteractor;

    public ChatsReaderPresenterImpl(ChatsAdapterView chatsAdapterView) {
        this.chatsView = chatsAdapterView;
        this.chatsReaderInteractor = new ChatsReaderInteractorImpl(this);
    }

    @Override
    public void subscribeForChatListUpdates() {
        chatsReaderInteractor.subscribeForUserChatsUpdates();
    }

    @Override
    public void unsubscribeForChatListUpdates() {
        chatsReaderInteractor.unsubscribeForUserChatsUpdates();
    }

    @Override
    public void onChatAdded(Chat chat) {
        chatsView.onChatAdded(chat);
    }

    @Override
    public void onChatChanged(Chat chat) {
        chatsView.onChatChanged(chat);
    }

    @Override
    public void onChatRemoved(String chatId) {
        chatsView.onChatRemoved(chatId);
    }
}

File: app/src/main/java/com/penseapp/acaocontabilidade/chat/chats/presenter/ChatsReaderPresenterImpl.kt
package com.penseapp.acaocontabilidade.chat.chats.presenter

import com.penseapp.acaocontabilidade.chat.chats.interactor.ChatsReaderInteractor
import com.penseapp.acaocontabilidade.chat.chats.interactor.ChatsReaderInteractorImpl
import com.penseapp.acaocontabilidade.chat.chats.model.Chat
import com.penseapp.acaocontabilidade.chat.chats.view.ChatsAdapterView

/**
 * Created by unity on 10/12/16.
 */

class ChatsReaderPresenterImpl(private val chatsView: ChatsAdapterView) : ChatsReaderPresenter {

    private val chatsReaderInteractor: ChatsReaderInteractor

    init {
        this.chatsReaderInteractor = ChatsReaderInteractorImpl(this)
    }

    override fun subscribeForChatListUpdates() {
        chatsReaderInteractor.subscribeForUserChatsUpdates()
    }

    override fun unsubscribeForChatListUpdates() {
        chatsReaderInteractor.unsubscribeForUserChatsUpdates()
    }

    override fun onChatAdded(chat: Chat) {
        chatsView.onChatAdded(chat)
    }

    override fun onChatChanged(chat: Chat) {
        chatsView.onChatChanged(chat)
    }

    override fun onChatRemoved(chatId: String) {
        chatsView.onChatRemoved(chatId)
    }
}
